assets/src/blocks/components/product-attributes/index.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Product attributes component.
+ */
+
+
+import { getProductAttributes } from '../utilities';
+import FastWCProductAttribute from './attribute'
+
+import PropTypes from 'prop-types';
+
+const { __ } = wp.i18n;
+const { useState, useEffect } = wp.element;
+
+const FastWCProductAttributes = ( {
+	onChange,
+	product,
+	variant,
+	selected,
+} ) => {
+	const [attKeys, setAttKeys] = useState([]);
+	const [labels, setLabels] = useState([]);
+	const [options, setOptions] = useState([]);
+
+	useEffect(
+		() => {
+			getProductAttributes( product, variant )
+				.then( ( list ) => {
+					const newOptions = {};
+					const newAttributes = {};
+
+					Object.keys( selected ).map( ( selectedKey ) => {
+						let availableValues = [];
+
+						if ( list.options && list.options[ selectedKey ] && list.options[ selectedKey ].length ) {
```

Comment:
should be able to use optional chaining here
```suggestion
						if ( list.options?.[ selectedKey ]?.length ) {
```